src/mlpack/methods/gmm/em_fit_impl.hpp

```diff
@@ -33,23 +35,36 @@ EMFit<InitialClusteringType, CovarianceConstraintPolicy>::EMFit(
     constraint(constraint)
 { /* Nothing to do. */ }
 
-template<typename InitialClusteringType, typename CovarianceConstraintPolicy>
-void EMFit<InitialClusteringType, CovarianceConstraintPolicy>::Estimate(
-    const arma::mat& observations,
-    std::vector<distribution::GaussianDistribution>& dists,
-    arma::vec& weights,
-    const bool useInitialModel)
+template<typename InitialClusteringType,
+         typename CovarianceConstraintPolicy,
+         typename Distribution>
+void EMFit<InitialClusteringType, CovarianceConstraintPolicy, Distribution>::
+Estimate(const arma::mat& observations,
+         std::vector<Distribution>& dists,
+         arma::vec& weights,
+         const bool useInitialModel)
 {
-  // Shortcut: if the user is using the DiagonalConstraint, then we will call
-  // out to Armadillo.  But Armadillo uses uword internally as an OpenMP index
-  // type, which crashes Visual Studio, so don't do this on Windows.
-  #ifndef _WIN32
-  if (std::is_same<CovarianceConstraintPolicy, DiagonalConstraint>::value)
+  if (std::is_same<Distribution,
+      distribution::DiagonalGaussianDistribution>::value)
   {
-    ArmadilloGMMWrapper(observations, dists, weights, useInitialModel);
-    return;
+    #ifdef _WIN32
+      Log::Warn << "Cannot use arma::gmm_diag on Visual Studio due to OpenMP"
+          << " compilation issues! Using slower EMFit::Estimate() instead..."
+          << std::endl;
+    #else
+      ArmadilloGMMWrapper(observations, dists, weights, useInitialModel);
+      return;
+    #endif
+  }
+  else if (std::is_same<CovarianceConstraintPolicy, DiagonalConstraint>::value
+      && std::is_same<Distribution, distribution::GaussianDistribution>::value)
+  {
+    // EMFit::Estimate() using DiagonalConstraint with GaussianDistribution
+    // uses slow implementation.
+    Log::Warn << "EMFit::Estimate() using DiagonalConstraint with "
+        << "GaussianDistribution uses slow implementation, so DiagonalGMM is "
```

Comment:
I think it should be written as `makes use of slower implementation ...` in place of `uses slow implementation`.